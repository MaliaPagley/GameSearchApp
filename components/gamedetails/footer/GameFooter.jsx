import React from 'react';
import { View, Text, Linking } from 'react-native';
import styles from './gamefooter.style';

const GameFooter = () => {
  return (
    <View testID="gameFooter" style={styles.container}>
      <Text testID="sourceLink" style={styles.link}
        onPress={() => Linking.openURL('https://rawg.io/')}>
          Source: Rawg.io
      </Text>
      <Text style={styles.dev}>App developed by Malia Pagley</Text>
    </View>
  )
}

export default GameFooter;