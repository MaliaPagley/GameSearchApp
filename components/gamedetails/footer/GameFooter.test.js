import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import GameFooter from './GameFooter';

jest.mock('react-native/Libraries/Linking/Linking', () => ({
  openURL: jest.fn(),
}));

describe('GameFooter Component', () => {
  it('component renders', () => {
    const { toJSON } = render(<GameFooter />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('handles press event correctly', () => {
    const { toJSON } = render(<GameFooter />);
    const link = screen.getByTestId('sourceLink');

    // Fire press event
    fireEvent.press(link);

    // Check if openURL function is called
    expect(require('react-native/Libraries/Linking/Linking').openURL).toHaveBeenCalled();

    // Snapshot testing
    expect(toJSON()).toMatchSnapshot();
  });
});

