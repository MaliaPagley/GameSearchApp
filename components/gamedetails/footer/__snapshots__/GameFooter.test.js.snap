// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameFooter Component component renders 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1A1F26",
      "bottom": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "left": 0,
      "padding": 12,
      "position": "absolute",
      "right": 0,
    }
  }
  testID="gameFooter"
>
  <Text
    onPress={[Function]}
    style={
      {
        "color": "#0066b2",
        "textAlign": "center",
      }
    }
    testID="sourceLink"
  >
    Source: Rawg.io
  </Text>
  <Text
    style={
      {
        "color": "#F3F4F8",
      }
    }
  >
    App developed by Malia Pagley
  </Text>
</View>
`;

exports[`GameFooter Component handles press event correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1A1F26",
      "bottom": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "left": 0,
      "padding": 12,
      "position": "absolute",
      "right": 0,
    }
  }
  testID="gameFooter"
>
  <Text
    onPress={[Function]}
    style={
      {
        "color": "#0066b2",
        "textAlign": "center",
      }
    }
    testID="sourceLink"
  >
    Source: Rawg.io
  </Text>
  <Text
    style={
      {
        "color": "#F3F4F8",
      }
    }
  >
    App developed by Malia Pagley
  </Text>
</View>
`;
